code/modules/mob/living/living.dm

```diff
@@ -715,40 +715,46 @@ Thanks.
-							var/list/damaged_organs = HM.drag_damage()
-							if (damaged_organs)
-								if(!HM.isincrit() && damaged_organs.len)
-									if(prob(HM.getBruteLoss() / 5)) //Chance to damage
-										for(var/datum/organ/external/damagedorgan in damaged_organs)
-											if((damagedorgan.brute_dam) < damagedorgan.max_damage)
-												HM.apply_damage(2, BRUTE, damagedorgan)
-												HM.visible_message("<span class='warning'>The wounds on \the [HM]'s [damagedorgan.display_name] worsen from being dragged!</span>")
-												HM.UpdateDamageIcon()
-									if(prob(HM.getBruteLoss() / 8)) //Chance to bleed
-										blood_splatter(HM.loc,HM)
-										var/blood_volume = round(HM:vessel.get_reagent_amount("blood"))
-										if(blood_volume > 0)
-											HM:vessel.remove_reagent("blood",4)
-											HM.visible_message("<span class='warning'>\The [HM] loses some blood from being dragged!</span>")
-
-								if(HM.isincrit() && damaged_organs.len) //Crit damage boost
-									if(prob(15))
-										for(var/datum/organ/external/damagedorgan in damaged_organs)
-											if((damagedorgan.brute_dam) < damagedorgan.max_damage)
-												HM.apply_damage(4, BRUTE, damagedorgan)
-												HM.visible_message("<span class='warning'>The wounds on \the [HM]'s [damagedorgan.display_name] worsen terribly from being dragged!</span>")
-												add_logs(src, HM, "caused drag damage to", admin = (M.ckey))
-												HM.UpdateDamageIcon()
-									if(prob(8))
+							var/list/damaged_organs = HM.get_broken_organs()
+							var/list/bleeding_organs = HM.get_bleeding_organs()
+							if (T.has_gravity() && HM.lying)
+							
+								if (damaged_organs.len)
+									if(!HM.isincrit())
+										if(prob(HM.getBruteLoss() / 5)) //Chance for damage based on current damage
+											for(var/datum/organ/external/damagedorgan in damaged_organs)
+												if((damagedorgan.brute_dam) < damagedorgan.max_damage) //To prevent organs from accruing thousands of damage
+													HM.apply_damage(2, BRUTE, damagedorgan)
+													HM.visible_message("<span class='warning'>The wounds on \the [HM]'s [damagedorgan.display_name] worsen from being dragged!</span>")
+													HM.UpdateDamageIcon()
+									else
+										if(prob(15))
+											for(var/datum/organ/external/damagedorgan in damaged_organs)
+												if((damagedorgan.brute_dam) < damagedorgan.max_damage)
+													HM.apply_damage(4, BRUTE, damagedorgan)
+													HM.visible_message("<span class='warning'>The wounds on \the [HM]'s [damagedorgan.display_name] worsen terribly from being dragged!</span>")
+													add_logs(src, HM, "caused drag damage to", admin = (M.ckey))
+													HM.UpdateDamageIcon()
+								
+								if (bleeding_organs.len && !(HM.species.anatomy_flags & NO_BLOOD))
+									var/blood_volume = round(HM:vessel.get_reagent_amount("blood"))
+									/*Sometimes species with NO_BLOOD get blood, hence weird check*/
```

Comment:
it's still checking for blood after checking for NO_BLOOD but yeah. I'll remove that comment in the upcoming massive change anyway.